### PR TITLE
fix(commerce): properly memoize suggestions controllers

### PR DESCRIPTION
**Pull request by Spuffynism**
Field suggestions controllers weren't being updated because we were memoizing their generation only on the list of facets, which did not change when field suggestions values changed. Now, we include the facet set (which includes facet values), to the function on which to memoize. This ensures the field suggestions controllers are update with the right values.

[CAPI-1102]

[CAPI-1102]: https://coveord.atlassian.net/browse/CAPI-1102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/features/commerce/facets/facet-search-set/commerce-facet-search-actions.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ const getExecuteFacetSearchThunkPayloadCreator =
             navigatorContext
           );
 
-    const response = await apiClient.facetSearch(await req);
+    const response = await apiClient.facetSearch(req);
```
Comment (Spuffynism) [inline packages/headless/src/features/commerce/facets/facet-search-set/commerce-facet-search-actions.ts:63]: This `await` had been forgotten. `req` is not a promise anymore.

**Review by alexprudhomme — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts [hunk 2]
```diff
@@ -79,10 +80,18 @@ export const commerceFacetSetReducer = createReducer(
     builder
       .addCase(fetchProductListing.fulfilled, handleQueryFulfilled)
       .addCase(executeSearch.fulfilled, handleQueryFulfilled)
-      .addCase(
-        executeCommerceFieldSuggest.fulfilled,
-        handleFieldSuggestionsFulfilled
+      .addCase(executeCommerceFieldSuggest.fulfilled, (state, action) =>
+        handleFieldSuggestionsFulfilled(state, action.payload.facetId)
       )
+      .addCase(fetchQuerySuggestions.fulfilled, (state, action) => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts:86]: We should cover this new case in facet-set-slice.test.ts!

File: packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts [hunk 3]
```diff
@@ -469,10 +478,8 @@ function handleQueryFulfilled(
 
 function handleFieldSuggestionsFulfilled(
   state: WritableDraft<CommerceFacetSetState>,
-  action: ReturnType<typeof executeCommerceFieldSuggest.fulfilled>
+  facetId: string
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts:481]: 👍

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts (hunk 2 above)
Comment (Spuffynism) [reply packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts:86]: Done in https://github.com/coveo/ui-kit/pull/4184/commits/37e44fa6a05d02f7fee52a9901239c6d52aad157

**Pull request merged**